pkg/config/config.go

```diff
@@ -936,3 +936,13 @@ func (c *Config) Write() error {
 	}
 	return nil
 }
+
+// Reload reloads the configuration from containers.conf files
+func Reload() (*Config, error) {
```

Comment:
Reading only the doc comments of `Reload()` and `Default()` it's not clear why we need that. What `Reload()` "secretly" does it to clear the cache, so we could have added a `ClearCache()` function and then used a `Default()` at the callers.

I have a slight preference for a `ClearCache()` or [InvalidateCache](https://github.com/containers/image/blob/master/pkg/sysregistriesv2/system_registries_v2.go#L434) as we do in c/image/pkg/sysregistriesv2.

@QiWang19 @rhatdan @TomSweeneyRedHat, WDYT? In case you prefer to stick with the current implementation, I would ask to only update the doc comments to make the caching (and invalidation) explicit in `Default()` and `Reload()`.

Reply:
Thanks for the suggestion. This is for the same purpose as InvalidateCache for long-running process. Are you suggesting first InvalidateCache, then recall Default(), will this fail to update the cache since the Default() is using sync.once?

Reply:
If we do a `configOnce = sync.Once{}`, we can call it again. That's a way to "reset" a `sync.Once`.

I added some more thoughts in https://github.com/containers/common/pull/219#pullrequestreview-448775849. As both functions could be used concurrently, we had to add `sync.Mutex` so we can retire the `sync.Once` entirely and use `sync.Mutex` to implement the singleton.

Default() would then look similar to:
```golang
func Default() (*Config, error) {
  mutex.Lock()
  defer mute.Unlock()
  if config != nil || err != nil {  // to replace the sync.Once
    return config, err
  }
  return NewConfig("")
}

Reply:
An InvalidateCache would set both to `nil` again

Reply:
@vrothberg thanks. PTAL https://github.com/containers/common/pull/224